MetaTools/buildUCD.py

```diff
@@ -75,20 +86,65 @@ def parse_range_properties(infile):
 
         ranges.append((first, last, data))
 
-    return ranges
+    ranges.sort()
+
+    # fill the gaps between explicitly defined ranges
+    last_start, last_end = -1, -1
+    full_ranges = []
+    for start, end, name in ranges:
+        assert last_end < start
+        assert start <= end
+        if start - last_end > 1:
+            full_ranges.append((last_end+1, start-1, default))
+        full_ranges.append((start, end, name))
+        last_start, last_end = start, end
+    if last_end != MAX_UNICODE:
+        full_ranges.append((last_end+1, MAX_UNICODE, default))
+
+    # reduce total number of ranges by combining continuous ones
+    last_start, last_end, last_name = full_ranges.pop(0)
+    merged_ranges = []
+    for start, end, name in full_ranges:
+        if name == last_name:
+            continue
+        else:
+            merged_ranges.append((last_start, start-1, last_name))
+            last_start, line_end, last_name = start, end, name
+    merged_ranges.append((last_start, MAX_UNICODE, last_name))
+
+    # make sure that the ranges cover the full unicode repertoire
+    assert merged_ranges[0][0] == 0
+    for (cs, ce, cn), (ns, ne, nn) in zip(merged_ranges, merged_ranges[1:]):
+        assert ce+1 == ns
+    assert merged_ranges[-1][1] == MAX_UNICODE
+
+    return merged_ranges
+
 
+def build_scripts(local_ucd=None, output_path=None):
+    """Fetch "Scripts.txt" data file from Unicode official website, parse
+    the script ranges and write them as a list of Python tuples to
```

Comment:
Now it writes two lists instead of a list of tuple

Reply:
I’ll fix the docstring, thanks for checking